aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -85,179 +87,71 @@ impl AggregatorChange {
                         delta: prev_delta,
                         history: prev_history,
                     } => {
-                        // Another useful macro, this time for merging deltas with different signs, such
-                        // as +A-B and -A+B. In these cases we have to check which of A or B is greater
-                        // and possibly flip a sign.
-                        macro_rules! update_different_sign {
-                            ($a:ident, $b:ident) => {
-                                if $a >= $b {
-                                    DeltaValue::Positive(subtraction($a, $b)?)
-                                } else {
-                                    DeltaValue::Negative(subtraction($b, $a)?)
-                                }
-                            };
-                        }
-
-                        let new_delta =
-                            match prev_delta {
-                                DeltaValue::Positive(prev_value) => match delta {
-                                    DeltaValue::Positive(self_value) => DeltaValue::Positive(
-                                        addition(prev_value, self_value, self.max_value)?,
-                                    ),
-                                    DeltaValue::Negative(self_value) => {
-                                        update_different_sign!(prev_value, self_value)
-                                    },
-                                },
-                                DeltaValue::Negative(prev_value) => match delta {
-                                    DeltaValue::Positive(self_value) => {
-                                        update_different_sign!(self_value, prev_value)
-                                    },
-                                    DeltaValue::Negative(self_value) => DeltaValue::Negative(
-                                        addition(prev_value, self_value, self.max_value)?,
-                                    ),
-                                },
-                            };
+                        let new_delta = prev_delta.add(&delta, self.max_value)?;
 
-                        // new_min_overflow = min(prev_min_overflow, prev_delta + min_overflow)
-                        let new_min_overflow = match (
-                            prev_history.min_overflow_positive_delta,
-                            history.min_overflow_positive_delta,
-                        ) {
-                            (
-                                Some(prev_min_overflow_positive_delta),
-                                Some(min_overflow_positive_delta),
-                            ) => {
-                                // We know that previous_speculative_value + prev_delta + min_overflow_positive_delta is
-                                // greater than max_value. Otherwise, valiate_history function should have returned an error.
```

Comment:
we are not calling validate_history above any more. so now we need to check min_overflow_positive_delta + prev_delta >= 0, and fail merging otherwise